### DISCUSSION TITLE
homodimer design using Rosetta Design, or fixbb, same mutation(s) between the two chains

### DISCUSSION THREAD

**Discussion by lanselibai** [RosettaCommons/rosetta#232] [General]
I would like to design a homodimer on Rosetta Design webserver (or using fixbb application), giving the same mutation between the two chains, Chain A and Chain B.
http://rosettadesign.med.unc.edu/submitjob.php

Is that possible to simply tell the program that the same mutations should be designed between the two chains? It seems that resfile could not achieve this, except using PIKAA to fix the target mutation.

Symmetry seems to be possible, but it looks like very complicated.
https://docs.rosettacommons.org/demos/latest/tutorials/Symmetry/Symmetry

**Comment by roccomoretti** [comment]
Symmetry is good if you want to match the structure between the two chains. It's not particularly complicated, once you create the symmdef file. (Which, if you already have a symmetric A/B complex should be straightforward.)

If you don't want to match the structures but only want to match the residue identities, then you can use the [LinkResidues](https://docs.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/TaskOperations/taskoperations_pages/LinkResidues) functionality. This is not accessible through the webserver or through the fixbb application, so you would need to create an XML file which uses it.

An alternative is using the [FavorSymmetricSequence](https://docs.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/Movers/movers_pages/FavorSymmetricSequenceMover) mover to add residue type constraints between the two chains. Again, you'd need to use RosettaScripts XML to access it.  (Unfortunately the underlying ResidueTypeLinkingConstraint is not accessible through the normal Rosetta constraint machinery.)
